### PR TITLE
Fix unittests with assertRaises + replace  assertTrue/False with better checks

### PR DESCRIPTION
**Pull request by kiblik**
Some `assert*` are skipped because they are evaluated in `with self.assertRaises(...)`. These asserts should be outside of these blocks.

Some unittests have been hard to troubleshoot if they use only `assertTrue` or `assertFalse`

### CODE REVIEW

**Review by Maffooch — APPROVED** [review]

File: unittests/tools/test_auditjs_parser.py
Comment (cneill) [inline unittests/tools/test_auditjs_parser.py:70]: Looks like this could be an `assertIn`, or no?

```suggestion
        self.assertIn(
            "Invalid JSON format. Are you sure you used --json option ?", str(context.exception)
        )
```

File: unittests/tools/test_checkmarx_osa_parser.py
Comment (cneill) [inline unittests/tools/test_checkmarx_osa_parser.py:193]: Looks like this should be an `assertIn`?

```suggestion
        self.assertIn(
            "Invalid format: missing mandatory field libraryId", str(context.exception)
        )
```

**Review by mtesauro — APPROVED** [review]
Approved

File: unittests/tools/test_kubehunter_parser.py
Comment (cneill) [inline unittests/tools/test_kubehunter_parser.py:46]: Did the contents of this exception change? Or was this maybe an error?

File: unittests/tools/test_sysdig_reports_parser.py
Comment (cneill) [inline unittests/tools/test_sysdig_reports_parser.py:49]: Should this be an `assertIn`? Or did the exception change?

```suggestion
        self.assertIn(
            "Number of fields in row (22) does not match number of headers (21)", str(context.exception)
        )
```

File: unittests/tools/test_sysdig_reports_parser.py
Comment (cneill) [inline unittests/tools/test_sysdig_reports_parser.py:62]: Should this be an assertIn? Or did the exception change?

```suggestion
        self.assertIn(
            "Number of fields in row (22) does not match number of headers (21)", str(context.exception)
        )
```

**Review by cneill — COMMENTED** [review]
Thanks for doing this. Looks like e.g. this spot was using an incorrect assertion previously

https://github.com/DefectDojo/django-DefectDojo/pull/9435/files#diff-6b013a76b51a339f8e86018f5ff1543b48734bf08abb7d467490b8d90743f162L36

A few questions/comments, and I just introduced some merge conflicts by merging a bunch of PRs if you could take a look at those.

File: unittests/tools/test_checkmarx_osa_parser.py
Comment (kiblik) [reply unittests/tools/test_checkmarx_osa_parser.py:193]: Both are valid. The question is, how strict are going to be?

File: unittests/tools/test_kubehunter_parser.py
Comment (kiblik) [reply unittests/tools/test_kubehunter_parser.py:46]: In my opinion, this part was copy-pasted from different unittests and never properly tested. It hasn't been evaluated because it was skipped (because of `ValueError` error).
Check this: https://github.com/DefectDojo/django-DefectDojo/actions/runs/7711228961/job/21016345466
```
2024-01-30T12:50:40.0190551Z [36muwsgi_1         |[0m ======================================================================
2024-01-30T12:50:40.0191264Z [36muwsgi_1         |[0m FAIL: test_kubehunter_parser_empty_with_error (unittests.tools.test_kubehunter_parser.TestKubeHunterParser.test_kubehunter_parser_empty_with_error)
2024-01-30T12:50:40.0191529Z [36muwsgi_1         |[0m ----------------------------------------------------------------------
2024-01-30T12:50:40.0191738Z [36muwsgi_1         |[0m Traceback (most recent call last):
2024-01-30T12:50:40.0192422Z [36muwsgi_1         |[0m   File "/app/unittests/tools/test_kubehunter_parser.py", line 46, in test_kubehunter_parser_empty_with_error
2024-01-30T12:50:40.0192587Z [36muwsgi_1         |[0m     self.assertIn(
2024-01-30T12:50:40.0193131Z [36muwsgi_1         |[0m AssertionError: 'KubeHunter report contains errors:' not found in 'Expecting value: line 1 column 1 (char 0)'
2024-01-30T12:50:40.0193257Z [36muwsgi_1         |[0m 
```

File: unittests/tools/test_sysdig_reports_parser.py
Comment (kiblik) [reply unittests/tools/test_sysdig_reports_parser.py:49]: Both are valid. The question is, how strict are going to be?

File: unittests/tools/test_sysdig_reports_parser.py
Comment (kiblik) [reply unittests/tools/test_sysdig_reports_parser.py:62]: Both are valid. The question is, how strict are going to be?

File: unittests/tools/test_checkmarx_osa_parser.py
Comment (cneill) [reply unittests/tools/test_checkmarx_osa_parser.py:193]: You're right, `assertEqual` is better so long as the comparison is correct. Just wanted to be double-check 😄

**Review by cneill — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Comment by mtesauro** [comment]
@kiblik We got the approvals - happy to merge once the conflicts are sorted out.

**Comment by mtesauro** [comment]
:+1: @kiblik

**Pull request merged**
